drivers/sensor/tdk/mpu6050/mpu6050.c

```diff
@@ -40,19 +40,24 @@ static void mpu6050_convert_gyro(struct sensor_value *val, int16_t raw_val,
 }
 
 /* see "Temperature Measurement" section from register map description */
-static inline void mpu6050_convert_temp(struct sensor_value *val,
+static inline void mpu6050_convert_temp(enum mpu6050_device_type device_type,
+					struct sensor_value *val,
 					int16_t raw_val)
 {
-	val->val1 = raw_val / 340 + 36;
-	val->val2 = ((int64_t)(raw_val % 340) * 1000000) / 340 + 530000;
-
-	if (val->val2 < 0) {
-		val->val1--;
-		val->val2 += 1000000;
-	} else if (val->val2 >= 1000000) {
-		val->val1++;
-		val->val2 -= 1000000;
-	}
+	int64_t tmp_val = (int64_t)raw_val * 1000000;
+
+	switch (device_type) {
+	case DEVICE_TYPE_MPU6500:
+		tmp_val = (tmp_val * 1000 / 333870) + 21000000;
+		break;
+
+	case DEVICE_TYPE_MPU6050:
+	default:
+		tmp_val = (tmp_val / 340) + 36000000;
```

Comment:
```suggestion
		tmp_val = (tmp_val / 340) + 36000000;
		break;
```

Reply:
This is the fall through, no break is needed

Reply:
e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/grp/fs_mgmt/src/fs_mgmt.c#L914

Reply:
You could use `__fallthrough` between the case statements to signal this to the compiler (and anybody reading this code).

Reply:
Fallthrough is not needed, if `DEVICE_TYPE_MPU6500` is used then the first case block runs and it breaks, if it is not then the default one runs and there's no more case statements, there is nothing to break from, it then falls through to the exit of the switch statement

Reply:
Yeah, I meant between `case DEVICE_TYPE_MPU6050:` and `default:`.

Reply:
My suggestion for a final break is based on style really, so I won't block this PR.

K&R V2 section 3.4:
_As a matter of good form, put a break after the last case (the default here) even though it's logically unnecessary.  Some day when another case gets added at the end, this bit of defensive programming will save you._